src/averageAges.js

```diff
@@ -55,7 +81,38 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const mothersArr = people.map(person => person['mother']);
+  const mothers = people.filter(person => mothersArr.includes(person['name']));
+  const men = people.filter(person => person['sex'] === 'm');
+  let countOfpeople = 0;
+  let result = 0;
+
+  const children = onlyWithSon
+    ? men.filter(y => y['mother']).filter(person => person !== null)
```

Comment:
you can get key from object by name, fix in all places
```suggestion
    ? men.filter(y => y.mother).filter(person => person !== null)
```